stdin-wrapper

```diff
@@ -6,15 +6,15 @@
 #   temporary file, and the first argument sets a particular file extension
 #   for the temporary file.
 
+set -x
+
 # All of the following arguments are read as command to run.
 file_extension="$1"
 shift
 
 temp_file=$(mktemp --tmpdir "ale-XXXXXXXXX$file_extension")
 trap 'rm $temp_file' EXIT
 
-while read -r; do
-    echo "$REPLY" >> "$temp_file"
-done
+cat - > "$temp_file"
```

Comment:
In theory it would be faster (mmaping and other fancy stuff). Also this allow me to to change it to named pipe soon.